projects/graph/graph.py

```diff
@@ -3,46 +3,58 @@
 """
 from util import Stack, Queue  # These may come in handy
 
+
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
+
     def __init__(self):
         self.vertices = {}
+
     def add_vertex(self, vertex):
-        """
-        Add a vertex to the graph.
-        """
-        pass  # TODO
+        self.vertices[vertex] = set()
+        # Sets are unique values, no duplicates
+
     def add_edge(self, v1, v2):
-        """
-        Add a directed edge to the graph.
-        """
-        pass  # TODO
```

Comment:
Good job breaking out your logic here Michael. It's _very_ clear what's happening by looking at each condition. How would you go about implementing this if you assumed _both_ `v1` and `v2` _are_ in `self.vertices`? If you were to skip that check you would have to handle the false condition in another way, but you'd also avoid making `add_edge` call on `add_vertex`.